Update to whinlatter release series and adapt to UNPACKDIR changes

alperak opened the pull request:
I haven’t done a full world build, but most recipes shouldn’t need further fixes. 

Related commits:

https://git.openembedded.org/openembedded-core/commit/?id=75eb26e71dba4096d5632b7f6b13db4f13aa6d7f
https://git.openembedded.org/openembedded-core/commit/?id=f80c07019ddadaf9c5fb890faabfda7920ecd15e
https://git.openembedded.org/openembedded-core/commit/?id=7321cc17ae5483f17fe9cdffea7b62acd9d9c3a2
https://git.openembedded.org/openembedded-core/commit/?id=f64b7e5fb3181734c8dde87b27e872a026261a74
https://git.openembedded.org/openembedded-core/commit/?id=46480a5e66747a673041fe4452a0ab14a1736d5e

alperak committed:
layer.conf: Update to whinlatter release series

Since there are currently multiple breaking changes, update to a new
release series for the next release so layers can show compatibility
correctly.

Signed-off-by: Alper Ak <alperyasinak1@gmail.com>

alperak committed:
remove S in recipes that fetch from git via setting BB_GIT_DEFAULT_DESTSUFFIX

Removing all the S = ${WORKDIR}/git assignments works because BB_GIT_DEFAULT_DESTSUFFIX
is set to match S from bitbake.conf (which itself is set to match typical tarball
releases).

https://git.openembedded.org/openembedded-core/commit/?id=f80c07019ddadaf9c5fb890faabfda7920ecd15e
https://git.openembedded.org/openembedded-core/commit/?id=46480a5e66747a673041fe4452a0ab14a1736d5e

Signed-off-by: Alper Ak <alperyasinak1@gmail.com>

alperak committed:
set S to be in UNPACKDIR in recipes that explicitly set S

https://git.openembedded.org/openembedded-core/commit/?id=46480a5e66747a673041fe4452a0ab14a1736d5e

Signed-off-by: Alper Ak <alperyasinak1@gmail.com>

alperak committed:
set S from UNPACKDIR in recipes that use only local files

There is no need to have two assignments; also setting S from UNPACKDIR (and not vice versa)
is more logical as unpack task comes first, and tasks that come after unpack use S to access unpack results.

Signed-off-by: Alper Ak <alperyasinak1@gmail.com>

nandra commented:
@alperak I'm getting following when try to build your branch:
```
/build_yocto_sunxi_master.sh
2025-06-23 07:51:22 - INFO     - kas 4.7 started
2025-06-23 07:51:22 - INFO     - Repository poky already contains b88519519fc59298ba6905453cb34d19b26ab794 as commit
2025-06-23 07:51:22 - INFO     - Repository oe already contains 820047afe48a0d48056f4752defc3d2803c9d906 as commit
2025-06-23 07:51:22 - INFO     - Repository sunxi already contains master as commit
2025-06-23 07:51:22 - INFO     - Repository meta-arm already contains 43d136216d5760734f6dd807b100fe82fd6c915e as commit
2025-06-23 07:51:22 - INFO     - Repository poky checked out to b88519519fc59298ba6905453cb34d19b26ab794
2025-06-23 07:51:23 - INFO     - Repository oe checked out to 820047afe48a0d48056f4752defc3d2803c9d906
2025-06-23 07:51:23 - INFO     - Repository sunxi checked out to f78f6ff2b7d182949c8441f9b974b25f8b12a6c1
2025-06-23 07:51:23 - INFO     - Repository meta-arm checked out to 43d136216d5760734f6dd807b100fe82fd6c915e
2025-06-23 07:51:23 - INFO     - /work/build$ /work/sources/poky/bitbake/bin/bitbake -c build core-image-minimal
Loading cache: 100% |                                                                                                                                                                                                         | ETA:  --:--:--
Loaded 0 entries from dependency cache.
WARNING: /work/build/../sources/meta-sunxi/recipes-bsp/u-boot/u-boot-sunxi.bb: /work/build/../sources/meta-sunxi/recipes-bsp/u-boot/u-boot-sunxi.bb:12 has a lack of whitespace around the assignment: 'DEFAULT_PREFERENCE_sun4i="1"'  0:00:09
WARNING: /work/build/../sources/meta-sunxi/recipes-bsp/u-boot/u-boot-sunxi.bb: /work/build/../sources/meta-sunxi/recipes-bsp/u-boot/u-boot-sunxi.bb:13 has a lack of whitespace around the assignment: 'DEFAULT_PREFERENCE_sun5i="1"'
WARNING: /work/build/../sources/meta-sunxi/recipes-bsp/u-boot/u-boot-sunxi.bb: /work/build/../sources/meta-sunxi/recipes-bsp/u-boot/u-boot-sunxi.bb:14 has a lack of whitespace around the assignment: 'DEFAULT_PREFERENCE_sun7i="1"'
WARNING: /work/build/../sources/meta-sunxi/recipes-bsp/u-boot/u-boot-sunxi.bb: /work/build/../sources/meta-sunxi/recipes-bsp/u-boot/u-boot-sunxi.bb:37 has a lack of whitespace around the assignment: 'SPL_BINARY="u-boot-sunxi-with-spl.bin"'
ERROR: ExpansionError during parsing /work/build/../sources/meta-openembedded/meta-oe/recipes-devtools/suitesparse/suitesparse_5.10.1.bb                                                                                       | ETA:  0:00:06
bb.data_smart.ExpansionError: Failure expanding variable CFLAGS:prepend[:=], expression was ${@" ".join(d.getVar('CC').split()[1:])}  which triggered exception AttributeError: 'NoneType' object has no attribute 'split'
The variable dependency chain for the failure is: CFLAGS:prepend[:=]

ERROR: Parsing halted due to errors, see error messages above
```

alperak committed:
Remove consecutive blank lines

Some of them were introduced by mass-removal of S = WORKDIR/git assignments;
rather than try to fix up (or redo) just these, I've run this sed command over
the whole tree:

sed -i -z -E 's/([ \t\f\v\r]*\n){3,}/\n\n/g' `find . -name *.bb -o -name *.inc`

The rationale is that more than one empty line is wasting vertical screen space, and
does nothing for readability.

Signed-off-by: Alper Ak <alperyasinak1@gmail.com>

nandra commented:
OK then lets wait until all is fixed in master so we have buildable master ;).

nandra commented:
Merged, thanks.
